trio/_core/_run.py

```diff
@@ -2745,7 +2746,7 @@ async def checkpoint_if_cancelled() -> None:
         task is task._runner.main_task and task._runner.ki_pending
     ):
         await _core.checkpoint()
-        assert False  # pragma: no cover
+        raise AssertionError()  # pragma: no cover
```

Comment:
very minor nit, but would be great with a message - any message, in the exception. A standard one I use when I don't come up with anything better is "this should never happen".

Reply:
Added reason text in a5afed7